### PR TITLE
advanced throwing spin grenades

### LINKED ISSUE #5567: Advanced Throwing throwable torque
Make grenades rotates in air, similar or better than in vanilla. Especially useful for WW2 German stick grenades. However some grenades have air stabilizer fins, so have to support both.

Opening this to discuss config entry requirements, so far thought of this (with ideas from below comments):
```cpp
class CfgAmmo {
    class SmokeShell;
    class ACE_G_Handflare_White: SmokeShell {
        // {} or {0,0,0} - disabled, {x, y, z} - torque vector
        ace_advanced_throwing_rotation = {100, 0, 0};
    };
};
```

It would also be enabled for all vanilla grenades with some default rotation, since they are all similar enough.

I didn't do any testing yet, but `addTorque` is applied as impulse, which means it should stop rotating relatively fast on very low torque vectors, thus simulating stabilizer fins.

### PR DESCRIPTION
**Pull request by commy2**
**When merged this pull request will:**
- title
- fix #5567

### CODE REVIEW

File: addons/advanced_throwing/XEH_postInit.sqf [hunk 1]
```diff
@@ -3,6 +3,17 @@
 // Fired XEH
 [QGVAR(throwFiredXEH), FUNC(throwFiredXEH)] call CBA_fnc_addEventHandler;
 
+// make grenades spin
+["ace_throwableThrown", { // local event
+    params ["_unit", "_grenade"];
+
+    private _config = _grenade call CBA_fnc_getObjectConfig;
+    private _torque = vectorNormalized getArray (_config >> QGVAR(torqueDirection)) vectorMultiply
+        getNumber (_config >> QGVAR(torqueMagnitude));
```
Comment (PabstMirror) [inline addons/advanced_throwing/XEH_postInit.sqf]: This works fine.
Has getNumber always worked on strings or was this something recent?
I thought I remember having to do some `call compile getText` to evaluate stuff like this.

Comment (commy2) [reply addons/advanced_throwing/XEH_postInit.sqf]: It worked like this since A2.

**Comment by jonpas** [comment]
I already did that: https://github.com/acemod/ACE3/compare/advThrowingRotation :P

I like this implementation more in terms of config entries, I didn't really think of randomness. However, I'd prefer to do `addTorque` directly in `fnc_throw.sqf`, attaching to an event in same component where we throw it looks weird and separates code for no reason.

**Comment by commy2** [comment]
I really like the modularity of this, but I guess if everything was made this way, a third of the computation time would go into frameworks.

**Comment by jonpas** [comment]
It's 3 lines, I think it's more readable and concise if in the function directly.

**Comment by commy2** [comment]
Put it in https://github.com/acemod/ACE3/compare/master...advThrowingRotation

**Pull request closed without merging**